torch/nested/_internal/sdpa.py

```diff
@@ -615,6 +615,41 @@ def _post_process_flash_output(out: torch.Tensor, og_size):
     return out
 
 
+def _autocast(
+    query: torch.Tensor,
```

Comment:
I think we also need to convert the attn_mask too if it is not the bool type attn_mask

Reply:
Good point, I forgot that it can be non-boolean

Reply:
apparently attn_mask is not supported yet in NJT - I added a check for that but no tests (the tests would fail for non-None masks)

https://github.com/pytorch/pytorch/blob/525fdc0f950c90d54cf69c1de5f10e741f43db76/torch/nested/_internal/sdpa.py#L65-L67